File: core/src/io/github/jerukan/game/gameunits/unitdata/GoldmineUnit.java
package io.github.jerukan.game.gameunits.unitdata;

import io.github.jerukan.game.Player;
import io.github.jerukan.game.gameunits.Unit;
import io.github.jerukan.game.gameunits.unitdata.unitactions.UnitAction;
import io.github.jerukan.util.Assets;

public class GoldmineUnit extends BaseUnit {

    public static final int GOLDMINE_DEFAULT_PRODUCTION = 100;

    public GoldmineUnit() {
        name = "goldmine";
        baseHealth = 6;

        baseUpkeep = 0;

        baseCost = 450;

        actions = new UnitAction[]{};

        type = Type.BUILDING;

        description = "Increases money production by $" + GOLDMINE_DEFAULT_PRODUCTION;
        setTexture(Assets.getTexture(Assets.wall));
    }

    @Override
    public boolean canBuildCondition(Player owner) {
        return true;
    }

    @Override
    public void onCreation(Unit self) {

    }

    @Override
    public void onDeath(Unit self) {

    }

    @Override
    public void onTurnStart(Unit self) {
        self.getOwner().setMoney(self.getOwner().getMoney() + GOLDMINE_DEFAULT_PRODUCTION);
    }

    @Override
    public void onTurnEnd(Unit self) {

    }
}

File: core/src/io/github/jerukan/game/gameunits/unitdata/GoldmineUnit.kt
package io.github.jerukan.game.gameunits.unitdata

import io.github.jerukan.game.Player
import io.github.jerukan.game.gameunits.Unit
import io.github.jerukan.game.gameunits.unitdata.unitactions.DismissAction
import io.github.jerukan.game.gameunits.unitdata.unitactions.UnitAction
import io.github.jerukan.util.Assets

class GoldmineUnit : BaseUnit() {
    init {
        name = "goldmine"
        baseHealth = 6

        baseUpkeep = 0

        baseCost = 450

        actions = arrayOf(DismissAction(this))

        type = BaseUnit.Type.BUILDING

        description = "Increases money production by $$GOLDMINE_DEFAULT_PRODUCTION"
        texture = Assets.getTexture(Assets.goldmine)
    }

    override fun canBuild(owner: Player): Boolean {
        return true
    }

    override fun onCreation(self: Unit) {

    }

    override fun onDeath(self: Unit) {

    }

    override fun onTurnStart(self: Unit) {
        self.owner.money = self.owner.money + GOLDMINE_DEFAULT_PRODUCTION
    }

    override fun onTurnEnd(self: Unit) {

    }

    companion object {

        val GOLDMINE_DEFAULT_PRODUCTION = 100
    }
}
